### PR TITLE
refactor: Use `Base` prefix for channel and interaction base classes

### PR DESCRIPTION
**Pull request by suneettipirneni**
**Please describe the changes this PR makes and why it should be merged:**

#8012 and #8027 introduced some confusion about the difference between `AnyInteraction` vs `Interaction` and `AnyChannel` vs `Channel`. This PR makes the union types `Interaction` and `Channel` by making the base classes of both channels and interactions include a `Base` prefix. As a result the `Any*` types were removed.

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR includes breaking changes (methods removed or renamed, parameters moved or removed)

### CODE REVIEW

**Review by SpaceEEC — CHANGES_REQUESTED** [review]
You missed a few `Channel` references in the docstrings:
https://github.com/suneettipirneni/discord.js/blob/f2cb3df06eaa66c71146fb36889225061297d067/packages/discord.js/src/client/Client.js#L121
https://github.com/suneettipirneni/discord.js/blob/f2cb3df06eaa66c71146fb36889225061297d067/packages/discord.js/src/managers/ChannelManager.js#L110
https://github.com/suneettipirneni/discord.js/blob/f2cb3df06eaa66c71146fb36889225061297d067/packages/discord.js/src/structures/CommandInteraction.js#L89
https://github.com/suneettipirneni/discord.js/blob/f2cb3df06eaa66c71146fb36889225061297d067/packages/discord.js/src/structures/GuildAuditLogsEntry.js#L51
https://github.com/suneettipirneni/discord.js/blob/f2cb3df06eaa66c71146fb36889225061297d067/packages/discord.js/src/structures/Invite.js#L172
https://github.com/suneettipirneni/discord.js/blob/f2cb3df06eaa66c71146fb36889225061297d067/packages/discord.js/src/structures/MessageMentions.js#L118
https://github.com/suneettipirneni/discord.js/blob/f2cb3df06eaa66c71146fb36889225061297d067/packages/discord.js/src/structures/MessageMentions.js#L182
https://github.com/suneettipirneni/discord.js/blob/f2cb3df06eaa66c71146fb36889225061297d067/packages/discord.js/src/util/Util.js#L448
https://github.com/suneettipirneni/discord.js/blob/f2cb3df06eaa66c71146fb36889225061297d067/packages/discord.js/src/util/Util.js#L451
https://github.com/suneettipirneni/discord.js/blob/f2cb3df06eaa66c71146fb36889225061297d067/packages/discord.js/src/util/Util.js#L455

**Review by kyranet — DISMISSED** [review]
There are many references in the documentation to `Channel` in camel-case, which would refer to a class. They can (and probably should) be lower cased so they imply any channel object.

Alternatively, we can define the JSDoc types and link them with `{@link Channel}`, same for `Interaction`.

Furthermore, I found a few more issues:

https://github.com/suneettipirneni/discord.js/blob/362a49759671dfff5890a78242f4329231fd99b2/packages/discord.js/src/structures/BaseInteraction.js#L32
https://github.com/suneettipirneni/discord.js/blob/362a49759671dfff5890a78242f4329231fd99b2/packages/discord.js/src/structures/MessagePayload.js#L89
https://github.com/suneettipirneni/discord.js/blob/362a49759671dfff5890a78242f4329231fd99b2/packages/discord.js/src/structures/MessagePayload.js#L280
https://github.com/suneettipirneni/discord.js/blob/362a49759671dfff5890a78242f4329231fd99b2/packages/discord.js/src/structures/interfaces/InteractionResponses.js#L23
https://github.com/suneettipirneni/discord.js/blob/362a49759671dfff5890a78242f4329231fd99b2/packages/discord.js/src/structures/interfaces/InteractionResponses.js#L36

I also spotted this one, since it's using the old name (`AnyChannel` was renamed to `Channel` in the typings). Probably could use the defined type instead of duplicating it:

https://github.com/suneettipirneni/discord.js/blob/362a49759671dfff5890a78242f4329231fd99b2/packages/discord.js/typings/index.test-d.ts#L1137

And last, should we use `BaseChannel` over `Channel` as well as `BaseInteraction` over `Interaction`? (Although the types are defined in the typings, they're not in JSDocs, we should probably update this).

**Comment by GravIos3** [comment]
@suneettipirneni Can you resolve the requested changes?
@vladfrangu @iCrawl @SpaceEEC Please review

File: packages/discord.js/src/structures/MessagePayload.js [hunk 1]
```diff
@@ -86,14 +88,14 @@ class MessagePayload {
...
   /**
-   * Whether or not the target is an {@link Interaction} or an {@link InteractionWebhook}
+   * Whether or not the target is an {@link BaseInteraction} or an {@link InteractionWebhook}
    * @type {boolean}
    * @readonly
    */
   get isInteraction() {
-    const Interaction = require('./Interaction');
+    const BaseInteraction = getBaseInteraction();
     const InteractionWebhook = require('./InteractionWebhook');
```
Comment (vladfrangu) [inline packages/discord.js/src/structures/MessagePayload.js:97]: Might as well change this too in a future PR

**Review by vladfrangu — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Pull request merged**